### DISCUSSION TITLE
How to increase vercel serverless function timeout with a NX project using NextJS with TRPC

### DISCUSSION THREAD

**Discussion by tanshunyuan** [vercel/community#4270] [Help] [answered]
### Summary

Hello, currently i've a monorepo setup using NX alongside NextJS with TRPC. After the recent post [here](https://vercel.com/changelog/serverless-functions-can-now-run-up-to-5-minutes), i've tried increase/decrease a function timeout to no avail.

Here's the configuration i've tried
`vercel.json`
```
{
  "functions": {
    "pages/api/trpc/[trpc].ts": {
      "maxDuration": 15
    }
  }
}

```

In this file: https://github.com/endalk200/create-t3-nx/blob/main/libs/api/src/lib/router/auth.ts

i've tried to add
```
export const maxDuration=300
```

but both options are not working, i'm wondering if this explicit limit increase is not possible due to trpc? 

### Example

https://github.com/endalk200/create-t3-nx

### Steps to Reproduce

nil

**Comment by rychlis** [comment]
Was struggling with this for the past day, finally i found a way: just add exported `config` variable in the trpc file (auth.ts in your case):
```
//All trpc functions can be run max for 300 seconds
export const config = {
  maxDuration: 300,
};

// export API handler
export default createNextApiHandler({
  router: appRouter,
  createContext: createTRPCContext,
});
```

Found it here and it works for me: https://stackoverflow.com/a/77516504

Seems like this is undocumented in Vercel though 🤦

**Reply by leerob** [reply, marked as answer]
We have docs here: https://vercel.com/docs/functions/configuring-functions/duration

Lmk if this helps!

**Reply by rychlis** [reply]
Hey AFAIK the docs only mention exporting variable named maxDuration in the file like this:
```
export const maxDuration = 5; // This function can run for a maximum of 5 seconds
```
However this doesn't work for me at all, I need to export a variable object `config` instead. 

Only other option mentioned in docs is to configure functions in `vercel.json`, I tried like 10 route configuration there (including `api/trpc/[trpc]`, `pages/api/**/*`, `**/*`, `api/**/*`) but this was all ignored and max duration was still 15s for all Serverless Functions. 

I'm using the next 14.0.4 with pages router, maybe this works different in app router? The repo created by @tanshunyuan is very similar to my setup - https://github.com/endalk200/create-t3-nx but if needed, I can spend some time later to create a new reproduction repo.

**Reply by rychlis** [reply]
Wow sorry for the confusion, I can now see there are separate code snippets in docs for /app and /pages router, the `config` object is properly documented there 🤦 

![Screenshot 2024-01-17 at 9 24 34](https://github.com/vercel/community/assets/1819747/680c6190-3450-4676-b38c-a8514f63bc7e)


Seems like I'm not the only one who missed it so hope it helps someone :)
